### PR TITLE
expose new onRangeRenderCompleted event

### PR DESCRIPTION
**Pull request by alanrenmsft**
need this event to do post rendering actions. for example: focus a cell.

### CODE REVIEW

File: components/js/slickGrid.ts [hunk 1]
```diff
@@ -533,16 +539,8 @@ export class SlickGrid implements OnChanges, OnInit, OnDestroy, AfterViewInit {
     }
 
     private renderGridDataRowsRange(startIndex: number, count: number): void {
-        let editor = this._grid.getCellEditor();
-        let oldValue = editor ? editor.getValue() : undefined;
-        let wasValueChanged = editor ? editor.isValueChanged() : false;
         this.invalidateRange(startIndex, startIndex + count);
-        let activeCell = this.activeCell;
```
Comment (alanrenmsft) [inline components/js/slickGrid.ts]: scenario: I edited a cell  and scroll away without moving the focus, when I scroll back my edits will be gone. this doesn't seem correct to me.

anyone remember why was this needed?

Comment (anthonydresser) [reply components/js/slickGrid.ts]: This looks like it only applies when you edit a cell that that is invalidated, this is trying to reapply the edit when slickgrid will discard that edit.

Comment (anthonydresser) [reply components/js/slickGrid.ts]: Not sure you need to remove the code.

Comment (alanrenmsft) [reply components/js/slickGrid.ts]: @anthonydresser Based on what I observed from the following scenario, i think the logic is wrong, I am not in office today, could you please give it a try?
1. open edit data window for a table with more than 1000 rows (e.g. HumanResources.Employee table in adventureworks database)
2. select 1000 in the max rows drop down in the toolbar and click run button
3. edit a cell in the first row and leave the focus inside the text edit
4. scroll all the way down to the end of grid and then scroll back to the top
Expected: the updated value should still be there
Actual: the change is reverted

Comment (anthonydresser) [reply components/js/slickGrid.ts]: I see the scenario you are talking about, but this code does not impact that scenario. What you are talking about is a separate thing. 

What this code is doing is if the user is editing row 99 and for some reason (this code was written awhile ago by another team) the async data provider attempts to invalidate the rows 50-100 (perhaps on slow servers?) this code attempts to do that invalidation then reapply the edit to row 99.

Comment (alanrenmsft) [reply components/js/slickGrid.ts]: I think you are right, the issue is a separate thing, I tested again, removing it doesn't actually help, I didn't scroll away too far that's why i was under the impression that this code is causing the problem.

but this code seems to be doing nothing, it is simply getting the current value (getValue()) from the editor and reapply it.

File: components/js/slickGrid.ts [hunk 2]
```diff
@@ -543,6 +549,7 @@ export class SlickGrid implements OnChanges, OnInit, OnDestroy, AfterViewInit {
                 editor.setValue(oldValue);
             }
         }
+        this.onRangeRenderCompleted.emit({ startRow: startIndex, endRow: startIndex + count });
```
Comment (anthonydresser) [inline components/js/slickGrid.ts]: This name seems a bit miss leading, there isn't really when a range is rendered, this is when we have completed getting the data in the range and we are simply informing slick grid of this fact. Is this what you intend for this event to be?

For example when the grid first renders we will have rows 0-50 in 1 window and 50-100 in another window. However slickgrid will have only rendered the visible rows, lets say 0-10. The user then scrolls down to show 20-50. Slickgrid will then render rows 20-50 but this function won't be called on those rows since we already had rows 20-50 in a window.

**Comment by alanrenmsft** [comment]
@anthonydresser Now that we have introduced a new event in SlickGrid, I am surfacing it in angular-slickgrid. Could you please merge the PR and create a new release? Thanks a lot.

**Comment by alanrenmsft** [comment]
@anthonydresser  actually hold on for a moment, i see an issue here. let me fix it.

**Comment by alanrenmsft** [comment]
@anthonydresser good to go now. thanks
